uint16  error_code
string 	message

